### PR TITLE
refactor: new factories

### LINKED ISSUE #41: Update factories for Laravel 8 and eliminate legacy factory support

### PR DESCRIPTION
**Pull request by mblarsen**

### CODE REVIEW

**Comment by mblarsen** [comment]
@cbj4074 spotted some bugs a long the way

The AttributeValue table has no timestamp columns so I disabled them in the model.

The new order test failed because the products where only `made` not `created`.

File: tests/Feature/Repositories/OrderRepositoryTest.php [hunk 1]
```diff
@@ -49,12 +50,13 @@ public function itCanUpdate()
...
-        $order = factory($this->getRepository()->modelClass())->create();
+        /** @var Order */
+        $order = $this->factory()->create();
 
-        $products = factory($this->product->modelClass(), 2)->create();
+        $products = $this->factory($this->product)->count(2)->create();
 
         $products->each(function ($product) use ($order) {
-            $order->products()->attach($product->id, ['price' => $product->price]);
+            $order->products()->attach($product, ['price' => $product->price]);
```
Comment (cbj4074) [inline tests/Feature/Repositories/OrderRepositoryTest.php:59]: Good catch. :)

File: tests/TestCase.php [hunk 2]
```diff
@@ -22,7 +23,13 @@ public function setUp(): void
...
         parent::setUp();
 
-        $this->withFactories(__DIR__.'/../database/factories');
+        FactoriesFactory::guessFactoryNamesUsing(function ($name) {
+            $modelName = (string) '\\IndieHD\\Velkart\\Database\\Factories\\'.
+                (class_basename($name)).
+                'Factory';
+
+            return $modelName::new();
+        });
```
Comment (cbj4074) [inline tests/TestCase.php:32]: This is clever! How did you even discover that this method exists in the `Illuminate\Database\Eloquent\Factories\Factory` class?

**Review by cbj4074 — APPROVED** [review]
Great work! There's a lot of tedium in these types of updates, so thank you for taking the time to work through them!

File: tests/TestCase.php (hunk 2 above)
Comment (mblarsen) [reply tests/TestCase.php:32]: I clicked go to source in my editor and checked the class :)

I'm wondering if it should just return the class name and not the `::new()` thing. 

Will have to check.

Comment (mblarsen) [reply tests/TestCase.php:32]: Not sure why it worked, but it should really be with just the class name returned. Will create a bug PR

Comment (cbj4074) [reply tests/TestCase.php:32]: Indeed. I wonder how this worked before. I just want to ensure that it's "doing what we think", but we can take that conversation over to the new PR.

**Pull request merged**